Fix AT003: acceptance test function name should include underscore

Check: [AT003](https://github.com/bflad/tfproviderlint/tree/master/passes/AT003)

```
./gitlab/data_source_gitlab_projects_test.go:14:6: AT003: acceptance test function name should include underscore
./gitlab/data_source_gitlab_projects_test.go:53:6: AT003: acceptance test function name should include underscore
```

We should fix these reported issues, and then remove this check from the list of ignored checks in the gitlabproviderlint make target.

To reproduce, run `go run ./gitlabproviderlint ./gitlab` (pending #432)

--- a/internal/provider/data_source_gitlab_projects.go
+++ b/internal/provider/data_source_gitlab_projects.go
@@ -212,7 +212,6 @@ var _ = registerDataSource("gitlab_projects", func() *schema.Resource {
 
 		ReadContext: dataSourceGitlabProjectsRead,
 
-		// lintignore: S006 // TODO: Resolve this tfproviderlint issue
 		Schema: map[string]*schema.Schema{
 			"max_queryable_pages": {
 				Description: "The maximum number of project results pages that may be queried. Prevents overloading your Gitlab instance in case of a misconfiguration.",
@@ -791,6 +790,9 @@ var _ = registerDataSource("gitlab_projects", func() *schema.Resource {
 							Computed:    true,
 							Elem: &schema.Schema{
 								Type: schema.TypeMap,
+								Elem: &schema.Schema{
+									Type: schema.TypeString,
+								},
 							},
 						},
 						"packages_enabled": {

--- a/internal/provider/data_source_gitlab_projects_test.go
+++ b/internal/provider/data_source_gitlab_projects_test.go
@@ -12,8 +12,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
-// lintignore: AT003 // TODO: Resolve this tfproviderlint issue
-func TestAccDataGitlabProjectsSearch(t *testing.T) {
+func TestAccDataGitlabProjects_search(t *testing.T) {
 	projectName := fmt.Sprintf("tf-%s", acctest.RandString(5))
 
 	resource.Test(t, resource.TestCase{
@@ -52,8 +51,7 @@ func TestAccDataGitlabProjectsSearch(t *testing.T) {
 	})
 }
 
-// lintignore: AT003 // TODO: Resolve this tfproviderlint issue
-func TestAccDataGitlabProjectsGroups(t *testing.T) {
+func TestAccDataGitlabProjects_groups(t *testing.T) {
 	projectName := fmt.Sprintf("tf-%s", acctest.RandString(5))
 	groupName := fmt.Sprintf("tf-%s", acctest.RandString(5))
 	parentGroupName := fmt.Sprintf("tf-%s", acctest.RandString(5))

